### PR TITLE
drpc: show kubeObjectProtectionTime in status

### PR DESCRIPTION
**Pull request by raghavendra-talur**
Signed-off-by: Raghavendra Talur <raghavendra.talur@gmail.com>

### CODE REVIEW

File: controllers/drplacementcontrol_controller.go [hunk 1]
```diff
@@ -1872,6 +1872,10 @@ func (r *DRPlacementControlReconciler) updateResourceCondition(
 		drpc.Status.LastGroupSyncBytes = vrg.Status.LastGroupSyncBytes
 	}
 
+	if vrg.Status.KubeObjectProtection.CaptureToRecoverFrom != nil {
+		drpc.Status.LastKubeObjectProtectionTime = &vrg.Status.KubeObjectProtection.CaptureToRecoverFrom.StartTime
```
Comment (ShyamsundarR) [inline controllers/drplacementcontrol_controller.go:1876]: Use time.DeepCopyInto() instead of address assignment here?

**Review by ShyamsundarR — COMMENTED** [review]
Overall change is fine, here are some things to consider:
- is `StartTime` updated each time backup is initiated? Because, if so this will always be current and not show up a delay in case backups are failing (we discussed this over a call)
- Protected condition may need an upgrade to consider if workload is protected or not based on this time and the `CaptureInterval` for k8s resources backup
- If we incorporate this into the Protected condition we would get the metric (and resulting alert) for free

File: controllers/drplacementcontrol.go [hunk 2]
```diff
@@ -2296,6 +2296,13 @@ func (d *DRPCInstance) shouldUpdateStatus() bool {
 		return true
 	}
 
+	if vrg.Status.KubeObjectProtection.CaptureToRecoverFrom != nil {
+		vrgKubeObjectProtectionTime := vrg.Status.KubeObjectProtection.CaptureToRecoverFrom.EndTime
```
Comment (ShyamsundarR) [inline controllers/drplacementcontrol.go:2300]: (nit) Is there a reason for a local variable here?

**Review by ShyamsundarR — COMMENTED** [review]
Mostly LGTM, just finishing up some additional checks in the code on when endTime would be set.

**Review by BenamarMk — APPROVED** [review]

File: controllers/drplacementcontrol.go (hunk 2 above)
Comment (raghavendra-talur) [reply controllers/drplacementcontrol.go:2300]: This is only to keep the line within 120 columns.

**Pull request merged**